From 8b09bb58efa3c392bdf17fc76620f6c29d0f11ce Mon Sep 17 00:00:00 2001
From: Vladimir Barinov <vladimir.barinov@cogentembedded.com>
Date: Fri, 4 Dec 2015 11:12:14 +0300
Subject: [PATCH] [media] rcar_vin: update state on stop streaming

This fixes the case when s/w and h/w streaming state
do not match. If hardware streaming is not stopped but
software state is STOPPED then this makes VIN still running.

Signed-off-by: Vladimir Barinov <vladimir.barinov@cogentembedded.com>
---
 drivers/media/platform/soc_camera/rcar_vin.c |    4 ++++
 1 file changed, 4 insertions(+)

Index: renesas-backport.view5/drivers/media/platform/soc_camera/rcar_vin.c
===================================================================
--- renesas-backport.view5.orig/drivers/media/platform/soc_camera/rcar_vin.c	2015-12-03 21:44:12.821881931 +0300
+++ renesas-backport.view5/drivers/media/platform/soc_camera/rcar_vin.c	2015-12-03 21:55:26.601869865 +0300
@@ -826,6 +826,10 @@
  */
 static void rcar_vin_wait_stop_streaming(struct rcar_vin_priv *priv)
 {
+	/* update the status if hardware is not stopped */
+	if (ioread32(priv->base + VNMS_REG) & VNMS_CA)
+		priv->state = RUNNING;
+
 	while (priv->state != STOPPED) {
 		/* issue stop if running */
 		if (priv->state == RUNNING)
